python/lsst/daf/butler/registries/sqlRegistry.py

```diff
@@ -36,6 +40,14 @@
 __all__ = ("SqlRegistryConfig", "SqlRegistry")
 
 
+@event.listens_for(Engine, "connect")
+def _enableForeignKeysOnSqlite3(dbapiConnection, connectionRecord):
+    if isinstance(dbapiConnection, SQLite3Connection):
+        cursor = dbapiConnection.cursor()
+        cursor.execute("PRAGMA foreign_keys=ON;")
```

Comment:
Do you need a `try/finally` here to ensure that `cursor.close()` is called? Or the internet gives the alternative form of:

```python
from contextlib import closing
with closing(self.connection.cursor()) as cur:
```

Reply:
Hmm, we don't want the connection to be closed at least (because the whole point of the listener is to do this on every opened connection). I'm not sure about the cursor. I can test though.